tests/composite_pk/test_filter.py

```diff
@@ -182,6 +182,30 @@ def test_filter_comments_by_pk_in(self):
                     Comment.objects.filter(pk__in=pks).order_by("pk"), objs
                 )
 
+    def test_filter_comments_by_pk_in_subquery(self):
+        self.assertSequenceEqual(
+            Comment.objects.filter(
+                pk__in=Comment.objects.filter(pk=self.comment_1.pk),
+            ),
+            [self.comment_1],
+        )
+        self.assertSequenceEqual(
+            Comment.objects.filter(
+                pk__in=Comment.objects.filter(pk=self.comment_1.pk).values(
+                    "tenant_id", "id"
+                ),
+            ),
+            [self.comment_1],
+        )
+        self.comment_2.integer = self.comment_1.id
```

Comment:
changing `self.comment_2.integer` like this changes it in every tests in this class

Reply:
It's not the case. All objects created assigned during `setUpTestData` are deep-copied and isolated from each `TestCase` instances that are used for test methods.

See [the docs](https://docs.djangoproject.com/en/5.1/topics/testing/tools/#django.test.TestCase.setUpTestData) about it.

Reply:
ohhh, I didn't know about that! 😄 apparently I've been avoiding doing this for years for the wrong reason